### PR TITLE
fix: [CDS-48637]: Instance stats gaps if there is any exception while fetching last snapshot time

### PR DESCRIPTION
**Pull request by TarunUba1**
## Describe your changes
Instead of sending a nulll value when exception occured while getting Latest Record. We will now be thowing exception in order to prevent data lose. explained in ticket 
## Testing
Screenshot 2022-12-22 at 6 41 28 PM
Screenshot 2022-12-22 at 6 41 52 PM
Screenshot 2022-12-22 at 6 41 57 PM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/41902)

### CODE REVIEW

File: 126-instance/src/main/java/io/harness/repositories/instancestats/InstanceStatsRepositoryImpl.java [hunk 1]
```diff
@@ -43,12 +44,14 @@ public InstanceStats getLatestRecord(String accountId, String orgId, String proj
         return parseInstanceStatsRecord(resultSet);
       } catch (SQLException ex) {
         if (totalTries == MAX_RETRY_COUNT) {
-          log.error("Error while fetching latest instance stats record", ex);
+          log.error("Error while fetching latest instance stats record after all retries");
+          throw ex;
```
Comment (abosii-harness) [inline 126-instance/src/main/java/io/harness/repositories/instancestats/InstanceStatsRepositoryImpl.java:48]: Aren't we losing this case the retries?

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Review by abosii-harness — APPROVED** [review]
> Reviewed 6 of 8 files at r1, 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41902)* status: Some of the checks are too old. 8 file(s) to review. 1 of 2 approvals obtained.

**Review by namantalaycha10 — APPROVED** [review]
> Reviewed 6 of 8 files at r1, 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41902)* status: Some of the checks are too old.

**Review by namantalaycha10 — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41902)* status: Some of the checks are too old.

**Review by namantalaycha10 — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41902)* status: 1 of 2 approvals obtained.

**Review by abosii-harness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41902)* status: :shipit: complete! ✓

**Pull request merged**